service/init.go

```diff
@@ -303,6 +309,27 @@ func (pcf *PCF) Exec(c *cli.Context) error {
 	return err
 }
 
+func (pcf *PCF) StartKeepAliveTimer(nfProfile models.NfProfile) {
+	KeepAliveTimerMutex.Lock()
+	defer KeepAliveTimerMutex.Unlock()
+	pcf.StopKeepAliveTimer()
+	if nfProfile.HeartBeatTimer == 0 {
+		// heartbeat timer value set to 60 sec
+		nfProfile.HeartBeatTimer = 60
+	}
+	logger.InitLog.Infof("Started KeepAlive Timer: %v sec", nfProfile.HeartBeatTimer)
+	//AfterFunc starts timer and waits for KeepAliveTimer to elapse and then calls pcf.UpdateNF function
+	KeepAliveTimer = time.AfterFunc(time.Duration(nfProfile.HeartBeatTimer)*time.Second, pcf.UpdateNF)
+}
+
+func (pcf *PCF) StopKeepAliveTimer() {
+	if KeepAliveTimer != nil {
```

Comment:
no lock while stopping timer ?

Reply:
calling from StartTimer.. so no use of lock